src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -1,5 +1,28 @@
 #include "ray/raylet/scheduling/cluster_resource_data.h"
 
+const std::string ResourceEnumToString(PredefinedResources resource) {
```

Comment:
Suggest using a static vector of the string labels, then use the enum as an index.

Reply:
Actually, ideally I think we should just use a protobuf enum, since it already has a method to convert to the string name.